src/indexer/indexer.utils.ts

```diff
@@ -0,0 +1,60 @@
+import { NotFound } from '@tsed/exceptions';
+import { Chain } from '../chains/config/chain.config';
+import { CachedSettSnapshot } from '../setts/interfaces/cached-sett-snapshot.interface';
+import { SettDefinition } from '../setts/interfaces/sett-definition.interface';
+import { SettSnapshot } from '../setts/interfaces/sett-snapshot.interface';
+import { getSett } from '../setts/setts.utils';
+import { getToken } from '../tokens/tokens-util';
+
+export const settToCachedSnapshot = async (chain: Chain, settDefinition: SettDefinition): Promise<CachedSettSnapshot> => {
+  const { settToken } = settDefinition;
+  const targetToken = getToken(settToken);
+  const { sett } = await getSett(chain.graphUrl, settToken);
+
+  if (!sett) {
+    // sett has not been indexed yet, or encountered a graph error
+    throw new NotFound(`${targetToken.name} sett not found`);
+  }
+
+  const { balance, totalSupply, pricePerFullShare, token } = sett;
+  const tokenBalance = balance / Math.pow(10, token.decimals);
+  const supply = totalSupply / Math.pow(10, token.decimals);
+  const ratio = pricePerFullShare / Math.pow(10, token.decimals);
+  const tokenPriceData = await chain.strategy.getPrice(token.id);
+  const value = tokenBalance * tokenPriceData.usd;
+
+  return Object.assign(new CachedSettSnapshot(), {
```

Comment:
That seems really weird. =)

Reply:
yeah... now that i got it working i'm gonna play around with that assignment. it was just form the docs - not a fan...